client/components/data/query-countries/index.jsx

```diff
@@ -0,0 +1,47 @@
+/**
+ * External dependencies
+ */
+import { Component, PropTypes } from 'react';
+import { connect } from 'react-redux';
+
+/**
+ * Internal dependencies
+ */
+import { isFetching } from 'state/countries/selectors';
+import { requestCountries } from 'state/countries/actions';
+
+class QueryCountries extends Component {
+	componentWillMount() {
+		this.request( this.props );
+	}
+
+	componentWillReceiveProps( nextProps ) {
+		if ( this.props.listType !== nextProps.listType ) {
+			this.request( nextProps );
+		}
+	}
+
+	request( props ) {
+		if ( ! props.isRequesting ) {
+			props.requestCountries( props.listType );
+		}
+	}
+
+	render() {
+		return null;
+	}
+}
+
+QueryCountries.propTypes = {
```

Comment:
Optional but this can be inlined with the component:
```javascript
class QueryCountries extends Component {
    static propTypes = {
        //...
    }
```